src/db/mysql/test.rs

```diff
@@ -837,3 +838,32 @@ fn delete_storage() -> Result<()> {
     assert_eq!(cid2, cid);
     Ok(())
 }
+
+#[test]
+fn lock_for_read() -> Result<()> {
+    let mut db = db()?;
+
+    let uid = 1;
+    let coll = "clients";
+    db.conn.transaction_manager().begin_transaction(&db.conn)?;
```

Comment:
I'll further flesh out the handling of transactions/locking later when we start integrating w/ handlers